app/assets/javascripts/discourse/app/components/modal/create-invite.gjs

```diff
@@ -0,0 +1,503 @@
+    });
+  }
+
+  get expireAfterOptions() {
+    let list = [1, 7, 30, 90];
+
+    if (!list.includes(this.siteSettings.invite_expiry_days)) {
+      list.push(this.siteSettings.invite_expiry_days);
+    }
+
+    list = list
+      .sort((a, b) => a - b)
+      .map((days) => {
+        return {
+          value: days,
+          text: I18n.t("dates.medium.x_days", { count: days }),
+        };
+      });
+
+    list.push({
+      value: 999999,
+      text: I18n.t("time_shortcut.never"),
+    });
+
+    return list;
+  }
+
+  @cached
+  get data() {
+    const data = {
+      restrictTo: this.invite.emailOrDomain ?? "",
+      maxRedemptions:
+        this.invite.max_redemptions_allowed ?? this.defaultRedemptionsAllowed,
+      inviteToTopic: this.invite.topicId,
+      inviteToGroups: this.model.groupIds ?? this.invite.groupIds ?? [],
+      customMessage: this.invite.custom_message ?? "",
+    };
+
+    if (this.inviteCreated) {
+      data.expiresAt = this.invite.expires_at;
+    } else {
+      data.expiresAfterDays = this.siteSettings.invite_expiry_days;
+    }
+
+    return data;
+  }
+
+  async save(data) {
+    let isLink = true;
+    let isEmail = false;
+
+    if (data.emailOrDomain) {
+      if (emailValid(data.emailOrDomain)) {
+        isEmail = true;
+        isLink = false;
+        data.email = data.emailOrDomain;
+      } else if (hostnameValid(data.emailOrDomain)) {
+        data.domain = data.emailOrDomain;
+      }
+      delete data.emailOrDomain;
+    }
+
+    if (isLink) {
+      if (this.invite.email) {
+        data.email = data.custom_message = "";
+      }
+    } else if (isEmail) {
+      if (data.max_redemptions_allowed > 1) {
+        data.max_redemptions_allowed = 1;
+      }
+
+      data.send_email = true;
+      if (data.topic_id) {
+        data.invite_to_topic = true;
+      }
+    }
+
+    this.saving = true;
+    try {
+      await this.invite.save(data);
+      const invites = this.model?.invites;
+      if (invites && !invites.any((i) => i.id === this.invite.id)) {
+        invites.unshiftObject(this.invite);
+      }
+
+      if (!this.simpleMode) {
+        this.flashText = sanitize(I18n.t("user.invited.invite.invite_saved"));
+        this.flashClass = "success";
+      }
+    } catch (error) {
+      this.flashText = sanitize(extractError(error));
+      this.flashClass = "error";
+    } finally {
+      this.saving = false;
+    }
+  }
+
+  get maxRedemptionsAllowedLimit() {
+    if (this.currentUser.staff) {
+      return this.siteSettings.invite_link_max_redemptions_limit;
+    } else {
+      return this.siteSettings.invite_link_max_redemptions_limit_users;
+    }
+  }
+
+  get defaultRedemptionsAllowed() {
+    const max = this.maxRedemptionsAllowedLimit;
+    const def = this.currentUser.staff ? 100 : 10;
+    return Math.min(max, def);
+  }
+
+  get canInviteToGroup() {
+    return (
+      this.currentUser.staff ||
+      this.currentUser.groups.any((g) => g.group_user?.owner)
```

Comment:
I think in this particular case it's OK because if the group isn't serialized, it means the user can't see it and wouldn't be able to select it from the groups dropdown which is only rendered if this `canInviteToGroup` property evals to true.

This would become a problem if in the future we decide to limit the number of groups we serialize in the `current_user` object, but at the moment we don't:

https://github.com/discourse/discourse/blob/16acba6cf85a91d8d6d73c691057df5117ed99df/app/serializers/current_user_serializer.rb#L101-L112

Reply:
Ah ok that's fine :+1: